go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -138,6 +138,7 @@ type QueryEngine struct {
 	mu               sync.RWMutex
 	tables           map[string]*schema.Table
 	plans            cache.Cache
+	streamPlans      cache.Cache
```

Comment:
my understanding is that the plan should not differ for stream / non-stream. 
We should have a single cache only.

Reply:
We should merge if there is any difference. The execution is different but the plan should be the same.

Reply:
Thanks for taking a look. The logic generating the [stream](https://github.com/vitessio/vitess/blob/main/go/vt/vttablet/tabletserver/planbuilder/plan.go#L263) plan is quite different from the [non-stream](https://github.com/vitessio/vitess/blob/main/go/vt/vttablet/tabletserver/planbuilder/plan.go#L205) plan generation - the non-stream plan generation uses GenerateLimitQuery vs the stream plan using GenerateFullQuery, the non-stream generation handling types of queries not handled by stream-plan generation, etc. I don't really have enough context to know if the resulting plans are significantly different or how the two logic branches could be merged.

Reply:
@harshit-gangal Just wanted to follow up on my question above

Reply:
My intension is to avoid doubling up the memory usage for cache.
We can add a prefix to SQL query as cache key to indicate it is a stream plan
(`__STREAM__`+<SQL>, Plan)

Reply:
Thanks for clarifying - I'll make that change.